custom_components/elasticsearch/es_publish_pipeline.py

```diff
@@ -160,47 +218,86 @@ async def async_init(self, config_entry: ConfigEntry) -> None:
-            await self._listener.async_init()
+            if self._settings.tags != []:
+                self._static_fields[CONF_TAGS] = self._settings.tags
+
+            if len(self._settings.change_detection_type) != 0:
+                await self._listener.async_init()
 
-            if self._settings.polling_enabled:
+            if self._settings.polling_frequency is not None and self._settings.polling_frequency > 0:
                 await self._poller.async_init(config_entry=config_entry)
+            else:
+                self._logger.debug("No polling frequency set. Disabling polling.")
 
             # Initialize document sinks
             await self._formatter.async_init(self._static_fields)
             await self._publisher.async_init(config_entry=config_entry)
 
-            filter_transform_load = LoopHandler(
-                name="es_filter_transform_load",
+            filter_format_publish = LoopHandler(
+                name="es_filter_format_publish_loop",
                 func=self._publish,
                 frequency=self._settings.publish_frequency,
                 log=self._logger,
             )
 
             self._cancel_publisher = config_entry.async_create_background_task(
                 self._hass,
-                filter_transform_load.start(),
-                "es_filter_transform_load",
+                async_create_catching_coro(filter_format_publish.start()),
+                "es_filter_format_publish_task",
             )
 
+            self._config_entry = config_entry
+
         async def _sip_queue(self) -> AsyncGenerator[dict[str, Any], Any]:
             while not self._queue.empty():
                 timestamp, state, reason = self._queue.get()
 
-                if not self._filterer.passes_filter(state, reason):
-                    continue
-
-                yield self._formatter.format(timestamp, state, reason)
+                try:
+                    yield self._formatter.format(timestamp, state, reason)
+                except Exception:
+                    self._logger.exception(
+                        "Error formatting document for entity [%s]. Skipping document.",
+                        state.entity_id,
+                    )
 
         async def _publish(self) -> None:
             """Publish the documents to Elasticsearch."""
+            try:
+                if not await self._gateway.check_connection():
+                    self._logger.debug("Skipping publishing as connection is not available.")
+                    return
+
+                await self._publisher.publish(iterable=self._sip_queue())
+
+            except AuthenticationRequired:
+                self._logger.error("Authentication issue in publishing loop. Reloading integration.")
+                self._logger.exception("Authentication issue in publishing loop.")
+
+                if self._config_entry:
+                    self._hass.config_entries.async_schedule_reload(self._config_entry.entry_id)
```

Comment:
I'm not sure where to put this, or if it's fixible or not -- If I fix the authorization error on the ES side, then HA still shows the integration entry as "Needs attention", and continues to prompt for credentials.

Steps:
1) Setup integration with API Key
2) Update privileges of API Key to no longer be sufficient (e.g. readonly)
3) Notice this component will sucessfully detect this, and reloads integration to trigger reauth flow
4) Update privileges of API Key so it is sufficient again
5) Notice this component starts ingesting docs again

Expected:
- Since docs are ingesting again, we should no longer prompt for new credentials

Actual:
- Prompt for new credentials remains in HA UI.

Reply:
Yeah, Im not sure if/how we can stop a started reconfigure flow

Reply:
That being said I could not reproduce this:

> Update privileges of API Key so it is sufficient again
> Notice this component starts ingesting docs again

When I did this, it actually doesn't start the integration? It runs through the config_flow, stops at API Key which wants to show a form and triggers the reconfigure but never starts the integration

Reply:
> That being said I could not reproduce this:
> 
> > Update privileges of API Key so it is sufficient again
> > Notice this component starts ingesting docs again
> 
> When I did this, it actually doesn't start the integration? It runs through the config_flow, stops at API Key which wants to show a form and triggers the reconfigure but never starts the integration

In my test, I didn't start/stop/reload the integration. The publish timer kicked off at the next interval and started succeeding after I fixed the API Key's privileges.

Reply:
Im not sure if the behavior changed since you tested but it should stop the integration now and the integration/publish loop should not start until the integration is reconfigured